server/app/invoice_handler.go

```diff
@@ -119,6 +121,64 @@ func (a *App) GetInvoiceHandler(req *http.Request) (interface{}, Response) {
 	}, Ok()
 }
 
+// DownloadInvoiceHandler downloads user's invoice by ID
+// Example endpoint: Downloads user's invoice by ID
+// @Summary Downloads user's invoice by ID
+// @Description Downloads user's invoice by ID
+// @Tags Invoice
+// @Accept  json
+// @Produce  json
+// @Security BearerAuth
+// @Param id path string true "Invoice ID"
+// @Success 200 {object} Response
+// @Failure 400 {object} Response
+// @Failure 401 {object} Response
+// @Failure 404 {object} Response
+// @Failure 500 {object} Response
+// @Router /invoice/download/{id} [get]
+func (a *App) DownloadInvoiceHandler(req *http.Request) (interface{}, Response) {
+	userID := req.Context().Value(middlewares.UserIDKey("UserID")).(string)
+
+	id, err := strconv.Atoi(mux.Vars(req)["id"])
+	if err != nil {
+		log.Error().Err(err).Send()
+		return nil, BadRequest(errors.New("failed to read invoice id"))
+	}
+
+	invoice, err := a.db.GetInvoice(id)
+	if err == gorm.ErrRecordNotFound {
+		return nil, NotFound(errors.New("invoice is not found"))
+	}
+	if err != nil {
+		log.Error().Err(err).Send()
+		return nil, InternalServerError(errors.New(internalServerErrorMsg))
+	}
+
+	if userID != invoice.UserID {
+		return nil, NotFound(errors.New("invoice is not found"))
+	}
+
+	// Get downloads dir
+	homeDir, err := os.UserHomeDir()
+	if err != nil {
+		log.Error().Err(err).Send()
+		return nil, InternalServerError(errors.New(internalServerErrorMsg))
+	}
+
+	downloadsDir := filepath.Join(homeDir, "Downloads")
```

Comment:
Can we make the full downloads home directory configurable and fallback to this if not defined, this allows the operator of the service to e.g mount external drives or use nfs directories .. etc